src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt

```diff
@@ -45,8 +43,8 @@ class GenericCatalogService(
    * Used to provide Catalog object to be used by other services internally.
    * Uses the cached catalog.
    */
-  fun getCatalogInternal(): Catalog {
-    return cachedCatalog
+  fun cachedServiceDefinitions(): List<ServiceDefinition> {
```

Comment:
f: Seems this method is not used anywhere in product code now. Can you check if the test code can also be refactored to not use this so that this can be removed. IMO better to now have unused public methods unless impossible to avoid it.

Reply:
Agree, though this requires quite a bit of a refactoring of the test code to include proper fixture builders, so I was shying away from that initially. but it's a good investment into the test code and enables writing future tests more quickly so will do that